queenbee/io/alias.py

```diff
@@ -0,0 +1,568 @@
+"""Queenbee alias input and output types.
+
+Use these alias inputs and outputs to create a different IO object for client side UIs.
+Alias inputs provide a handler object to convert the input provided to alias to what is
+required with the original input.
+"""
+
+import os
+from typing import Dict, Union, List
+
+from pydantic import constr, Field, validator
+from jsonschema import validate as json_schema_validator
+
+from .common import ItemType, GenericInput, FromOutput, find_dup_items
+from .artifact_source import HTTP, S3, ProjectFolder
+from ..io.reference import FileReference, FolderReference, TaskReference
+from ..base.basemodel import BaseModel
+
+
+class IOAliasHandler(BaseModel):
+    """Input and output alias handler object."""
+
+    type: constr(regex='^IOAliasHandler$') = 'IOAliasHandler'
+
+    language: str = Field(
+        ...,
+        description='Declare the language (e.g. python, csharp, etc.). This '
+        'option allows the recipe to be flexible on handling different programming '
+        'languages.'
+    )
+
+    module: str = Field(
+        ...,
+        description='Target module or namespace to load the alias function.',
+        example='honeybee_rhino.handlers'
+    )
+
+    function: str = Field(
+        ...,
+        description='Name of the function. The input value will be passed to this '
+        'function as the first argument.'
+    )
+
+
+class DAGGenericInputAlias(GenericInput):
+    """Base class for DAG Alias inputs.
+
+    This class adds a handler to input to handle the process of loading the input
+    from different graphical interfaces.
+    """
+    type: constr(regex='^DAGGenericInputAlias$') = 'DAGGenericInputAlias'
+
+    platform: List[str] = Field(
+        ...,
+        description='Name of the client platform (e.g. Grasshopper, Revit, etc). The '
+        'value can be any strings as long as it has been agreed between client-side '
+        'developer and author of the recipe.'
+    )
+
+    handler: List[IOAliasHandler] = Field(
```

Comment:
Renaming the key to handlers?

Reply:
I think we have kept most of the keys singular but let me check - I'm totally fine with changing this to handlers.